qiita_db/test/test_artifact.py

```diff
@@ -0,0 +1,374 @@
+        self.assertEqual(obs.ebi_run_accessions, dict())
+        self.assertEqual(obs.study, Study(1))
+
+    def test_delete_error_public(self):
+        test = Artifact.create(self.filepaths_root, "FASTQ",
+                               prep_template=self.prep_template)
+        test.visibility = "public"
+        with self.assertRaises(QiitaDBArtifactDeletionError):
+            Artifact.delete(test.id)
+
+    def test_delete_error_has_children(self):
+        with self.assertRaises(QiitaDBArtifactDeletionError):
+            Artifact.delete(1)
+
+    def test_delete_error_analyzed(self):
+        with self.assertRaises(QiitaDBArtifactDeletionError):
+            Artifact.delete(4)
+
+    def test_delete_error_ebi(self):
+        obs = Artifact.create(self.filepaths_processed, "Demultiplexed",
+                              parents=[Artifact(1)],
+                              processing_parameters=Parameters(1, Command(1)),
+                              can_be_submitted_to_ebi=True,
+                              can_be_submitted_to_vamps=True)
+        obs.ebi_run_accessions = {'1.SKB1.640202': 'ERR1000001',
+                                  '1.SKB2.640194': 'ERR1000002'}
+        with self.assertRaises(QiitaDBArtifactDeletionError):
+            Artifact.delete(obs.id)
+
+    def test_delete_error_vamps(self):
+        obs = Artifact.create(self.filepaths_processed, "Demultiplexed",
+                              parents=[Artifact(1)],
+                              processing_parameters=Parameters(1, Command(1)),
+                              can_be_submitted_to_ebi=True,
+                              can_be_submitted_to_vamps=True)
+        obs.is_submitted_to_vamps = True
+        with self.assertRaises(QiitaDBArtifactDeletionError):
+            Artifact.delete(obs.id)
+
+    def test_delete(self):
+        test = Artifact.create(self.filepaths_root, "FASTQ",
+                               prep_template=self.prep_template)
+
+        Artifact.delete(test.id)
+
+        with self.assertRaises(QiitaDBUnknownIDError):
+            Artifact(test.id)
+
+    def test_tiemstamp(self):
+        self.assertEqual(Artifact(1).timestamp,
+                         datetime(2012, 10, 1, 9, 30, 27))
+        self.assertEqual(Artifact(2).timestamp,
+                         datetime(2012, 10, 1, 10, 30, 27))
+        self.assertEqual(Artifact(3).timestamp,
+                         datetime(2012, 10, 1, 11, 30, 27))
+        self.assertEqual(Artifact(4).timestamp,
+                         datetime(2012, 10, 2, 17, 30, 00))
+
+    def test_processing_parameters(self):
+        self.assertIsNone(Artifact(1).processing_parameters)
+        self.assertEqual(Artifact(2).processing_parameters,
+                         Parameters(1, Command(1)))
+        self.assertEqual(Artifact(3).processing_parameters,
+                         Parameters(2, Command(1)))
+
+    def test_visibility(self):
```

Comment:
This and test_artifact_type seem not idea. Basically, they are testing that the data created in the populate is right (testing getter) but not that the setters/getters work as expected.